src/services/lnd/onchain-service.ts

```diff
@@ -25,19 +25,20 @@ export const OnChainService = (
     try {
       const { current_block_height } = await getHeight({ lnd })
       const after = Math.max(0, current_block_height - scanDepth) // this is necessary for tests, otherwise after may be negative
-      const { transactions } = await getChainTransactions({ lnd, after })
+      const { transactions }: GetChainTransactionsResult = await getChainTransactions({
+        lnd,
+        after,
+      })
 
       return transactions
-        .filter((tx) => !tx.is_outgoing || !!tx.transaction)
-        .map((tx) => {
+        .filter((tx) => !tx.is_outgoing || !!tx.transaction || !!tx.fee)
```

Comment:
if this method is only for incoming transactions then filter should be something like `!tx.is_outgoing && (tx.transaction && tx.fee >= 0`

Reply:
Added a unit test to handle all cases.